tests/conftest.py

```diff
@@ -148,23 +146,51 @@ def __init__(self, *args, **kwargs):
         # Makes it easier to work with the bot in tests
         self._unfreeze()
 
+    # Here we override get_me for caching because we don't want to call the API repeatedly in tests
+    async def get_me(self, *args, **kwargs):
+        if self._bot_user is None:
+            self._bot_user = make_mock_user(self.token)
+        return self._bot_user
+
 
 class DictBot(Bot):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         # Makes it easier to work with the bot in tests
         self._unfreeze()
 
+    # Here we override get_me for caching because we don't want to call the API repeatedly in tests
+    async def get_me(self, *args, **kwargs):
+        if self._bot_user is None:
+            self._bot_user = make_mock_user(self.token)
+        return self._bot_user
+
 
 class DictApplication(Application):
     pass
 
 
+def make_mock_user(token: str) -> User:
+    """Used to return a mock user in bot.get_me(). This saves API calls on every init."""
+    bot_info = get_bot()
+    user_id = int(token.split(":")[0] or bot_info["token"].split(":")[0])
```

Comment:
what is the `or` needed for? in which cases do we have `token.split(":")[0] is None`?

Reply:
okay, I was confused, b/c the argument isn't even optional. But if `get_bot` already contains the token, then we can drop the parameter alltogether, right? In that case I would like to move this function into the `bots.py` file (since it only relies on the info that we load from the encoded data there) and rename it to maybe `get_bot_user`

Reply:
I remember at least one test (test_eq in test_bot.py) where we did something like: `make_bot(FALLBACKS[0]["token"])` so in that case, we'll need the parameter so we can pass the token explicitly instead of getting one from `get_bot()`. I'll rename the function regardless

Reply:
`make_bot` is a different function than `make_mock_user` though, isn't it? The token isn't even used to get the user info of the bot with that token …

Reply:
> make_bot is a different function than make_mock_user though, isn't it

yes, but since it creates a new DictExtBot, `make_mock_user` is used in its `get_me`. 

`make_bot(FALLBACKS[0]["token"]).id != make_bot(FALLBACKS[1]["token"])` only because of the different user id which is fetched from the token. This important for the equality test

Reply:
okay, so the point is that we need to be able to make a bot with a specific id, but currently have no way (or are not willing to provide a way) of getting the correct user info from a token. So we're okay building a bot with the specified token & id, but username, first name etc not matching these token & id.
If you can add a CSI style comment about that, I'm good :)